app/bot/openaihistory_test.go

```diff
@@ -0,0 +1,59 @@
+package bot
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_LimitedMessageHistory(t *testing.T) {
```

Comment:
nit: i don't thing this function even need a test, especially such a complicated test. I don't mind it, just fyi